### PR TITLE
Added Type Annotations to sphinx-intl for PEP 561 compliance.

### PR DESCRIPTION
**Pull request by SekouD**
Hi,

I am the author of the python package [potranslator](https://pypi.org/project/potranslator/).

It is a utility to automatically translate po and pot files using Google Translate.

As I used to use sphinx-intl previously to translate my own projects, I was inspired by sphinx-intl's command line interface to create potranslator. As a result I adapted the code you use in commands.py to use in my utility, as well as the documentation for the command line interface.

To thank you for sharing sphinx-intl as an open source project, I decided to to contribute back to sphinx-intl by adding type annotations following the new [PEP 561 Distributing and Packaging Type Information](https://www.python.org/dev/peps/pep-0561/). This allows static type checkers like [mypy](http://mypy-lang.org/) to analyze the code for type errors before runtime or Integrated Development Environments  like [Pycharm](https://www.jetbrains.com/help/pycharm/type-hinting-in-product.html) to provide useful type hints and code completion for python.

You can check  that the type information is correct by installing mypy and while in the root folder of sphinx-intl running the following command:

mypy --strict --ignore-missing-import ./

The Pull Request contains all the necessary *.pyi files as well as the py.typed file.

If you want to use this contribution, you just have to build your distribution and everything will work.

I hope you will find this useful.

Cheers,

SekouD.

### CODE REVIEW

**Comment by shimizukawa** [comment]
Thanks you for your contribution, and sorry for my late reply.
Now I'm trying your change.

**Comment by shimizukawa** [comment]
With options as same as sphinx,
https://github.com/sphinx-doc/sphinx/blob/50cf7842f62221a501afaff3740a23986c6eacf0/setup.cfg#L45-L54

these errors are occurred:
```
sphinx_intl/commands.pyi: note: In class "LanguagesType":
sphinx_intl/commands.pyi:60:5: error: Incompatible types in assignment (expression has type "unicode", base class "_ParamType" defined the type as "str")
sphinx_intl/commands.pyi:61:5: error: Incompatible types in assignment (expression has type "unicode", base class "_ParamType" defined the type as "Optional[str]")
```

Without `python_version = 2.7`, I don't get such errors.

Do you have any str/unicode strategy for mypy with using Py2?

**Comment by SekouD** [comment]
Hi Takauki-san.

I will have a look at the issue today or tomorrow and come back to you as
soon as possible.

Have a nice day.

File: sphinx_intl/commands.pyi [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+option_language: Callable
+option_transifex_username: Callable
+option_transifex_password: Callable
+option_transifex_project_name: Callable
+CONTEXT_SETTINGS: Mapping[Text, Sequence[Text]]
+
+
+class LanguagesType(click.ParamType):
+    name: Union[Text, unicode]= ...
+    envvar_list_splitter: Union[Text, unicode]= ...
```
Comment (shimizukawa) [inline sphinx_intl/commands.pyi]: Unfortunately, this expression is interpreted as `Union[unicode, unicode]`.

So I've changed these 2 lines in my local as just `str`, mypy testing has been passed.

Comment (shimizukawa) [reply sphinx_intl/commands.pyi]: In order to execute mypy in CI test, please add below part in `setup.cfg` and `tox.ini` that copied from sphinx project.

`setup.cfg`:
```
[mypy]
python_version = 2.7
show_column_numbers = True
show_error_context = True
ignore_missing_imports = True
follow_imports = skip
incremental = True
check_untyped_defs = True
warn_unused_ignores = True
strict_optional = False
```

tox.ini:
```
envlist=py36,py35,py34,py27,flake8,sdist,mypy

...

[testenv:mypy]
basepython = python3
description =
    Run type checks.
deps =
    mypy
commands=
    mypy sphinx_intl/
```

**Comment by shimizukawa** [comment]
Now this PR is obsoleted I think.
Feel free to reopen if you can update. Thanks.

**Pull request closed without merging**